### PR TITLE
Added Instagram Metadata Scrapper

### LINKED ISSUE #441: Instagram Metadata Scrapper
# Description

Added script to scrape instagram metadata of a profile , just enter the username and get the metadata like follower count , profile picture url, follows_count and media count as python dict .

## Type of issue
- NOTE: *These boxes can be checked using **`[X]`***

- [X] Feature (New Script)
- [ ] Bug
- [ ] Documentation

## Checklist:

- [X] I have read the project guidelines.
- [X] I have checked all the existing projects, before submitting a new project issue.
- [X] I have checked previous issues to avoid duplicates.
- [X] This issue will be meaningful for the project.

### PR DESCRIPTION
**Pull request by AnkDos**
# Description

Added script to scrape instagram metadata of a profile , just enter the username and get the metadata like follower count , profile picture url, follows_count and media count as python dict .

Fixes #441

Replace `issue_no` in the above line, with the issue related to this PR.

## Type of change

Choosing one or more options from the following as per the nature of your Pull request.
- NOTE: *These boxes can be checked using **`[X]`***

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation Update

# Checklist:
Please tick all the boxes that are fulfilled by your Pull Request.

- [X] I have named my files and folder, according to this project's guidelines.
- [X] My code follows the style guidelines of this project.
- [X] My Pull Request has a descriptive title. (not a vague title like `Update index.md`)
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] I have created a helpful and easy to understand `README.md`, according to the given [`README_TEMPLATE.`](https://github.com/Python-World/Python_and_the_Web/blob/master/README_TEMPLATE.md)
- [X] I have included a requirements.txt file (if external libraries are required.)
- [X] My changes do not produce any warnings.
- [X] I have added a working sample/screenshot of the script.

### CODE REVIEW

**Comment by AdityaJ7** [comment]
@AnkDos  which issue is this related to ??

**Comment by AnkDos** [comment]
Hi @AdityaJ7  , it's a new feature to scrape Instagram profile meta data like followers count , following counts , media count and profile picture by entering the Instagram id.

**Comment by tusharnankani** [comment]
Which issue number is this? Is it present [here](https://github.com/Python-World/Python_and_the_Web/issues?q=is%3Aopen+is%3Aissue+)? @AnkDos 

If yes, refer it using `#issue_number`.

Else, create one, and ask for it to be assigned.

**Comment by AnkDos** [comment]
Done buddy @tusharnankani

File: Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import json
+import pprint
+import requests
+
+from bs4 import BeautifulSoup
+from faker import Faker
```
Comment (tusharnankani) [inline Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py:6]: Add `faker` to requirements.txt

File: Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                "profile_picture_url": ret_data['user'].get('profile_pic_url_hd', None),
+                "follower_count": ret_data['user'].get('edge_followed_by', {}).get('count',None),
+                "media_count": ret_data['user'].get('edge_owner_to_timeline_media', {}).get('count',0),
+                "follows_count": ret_data['user'].get('edge_follow',{}).get('count',None)
+            }
+        return {"error" : "problem getting the data , please try again"}
+
+    except Exception as _unused:
+        return {"error": "internal error"}
+pprint.pprint(get_metadata("tomhardy"))
```
Comment (tusharnankani) [inline Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py:37]: Add a comment asking the user to replace the username, or even better, ask for string input() for the username.

Comment (tusharnankani) [inline Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py:37]: Also, place it under a `if __name__ == "__main__:"`

Ask for input for username.

And then just add the following print line.

**Review by tusharnankani — CHANGES_REQUESTED** [review]
Here are some suggestions and changes:
- Comment your code for better understanding.
- Add a sample screenshot of the working script, in the README.md file.

File: Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py (hunk 1 above)
Comment (AnkDos) [reply Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py:6]: Added @tusharnankani

File: Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py (hunk 2 above)
Comment (AnkDos) [reply Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py]: Added @tusharnankani

Comment (AnkDos) [reply Scripts/Web_Scrappers/Instagram_Metadata_Scrapper/scrapper.py]: Added @tusharnankani

**Review by tusharnankani — COMMENTED** [review]
Screenshot (644)

Not working for me.

**Comment by AnkDos** [comment]
working_here
@tusharnankani  seems to be working here , maybe a setup issue on your machine . To check that you can print _unused variable in the except block .

**Comment by AdityaJ7** [comment]
@AnkDos  even I am facing the same issue as @tusharnankani, I checked it on two different platforms
fail

**Comment by AnkDos** [comment]
@AdityaJ7 @tusharnankani  Please do me a favour and print the _unused variable below line 37 inside except block before return statement and share the screenshot . Problem is not clear as its not throwing error on my system .Thanks

**Comment by AdityaJ7** [comment]
error

**Comment by AnkDos** [comment]
Screenshot from 2020-10-09 19-52-11
@tusharnankani @AdityaJ7  I tried on my different machine as well which is running arch linux . Cloned it , installed the requirements and ran the script and it went well . I will check for this decode error on windows machine and will get back . Thanks for sharing the screenshot .

**Comment by AnkDos** [comment]
@AdityaJ7 Buddy I don't have an access to a windows machine right now . In the latest commit I have dropped faker , assuming it might have been generating a user agent for which response from instagram is not desirable or what I am expecting and parsing . Can you please check now and see if its working on your machine ?

**Comment by tusharnankani** [comment]
> Can you please check now and see if its working on your machine ?

Still not working for me!

**Review by arghaooo — APPROVED** [review]

**Comment by AdityaJ7** [comment]
@AnkDos ok then leave it as it is and adjust your readme to include that - this script runs well only on Linux.
We'll try to find a solution and will update it later

**Comment by AnkDos** [comment]
@AdityaJ7 Done

**Review by tusharnankani — APPROVED** [review]

**Review by AdityaJ7 — APPROVED** [review]

**Comment by AdityaJ7** [comment]
@all-contributors please add @AnkDos for code

**Pull request merged**